freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -296,9 +303,15 @@ public Service deleteService(String canonicalPrincipal) throws FreeIpaClientExce
     }
 
     public RPCResponse<Service> addService(String canonicalPrincipal) throws FreeIpaClientException {
-        RPCResponse<Service> response = null;
-        //TODO Implement as part of CDPSDX-584
-        return response;
+        List<String> flags = List.of(canonicalPrincipal);
+        Map<String, Object> params = Map.of("force", true);
+        return invoke("service_add", flags, params, Service.class);
```

Comment:
is there any reason that here we return the whole RPCRespone instead of the result object? The same for all these new methods

Reply:
Good point. This really cleans up the code. Thanks!